### DISCUSSION TITLE
Why use invisible unicode character u+200d(zero-width joiner)?

### DISCUSSION THREAD

**Discussion by Garfield550** [endojs/endo#1726] [Q&A] [answered]
https://github.com/endojs/endo/blob/b288b593b87eff6054c1e94122e24c6e9a79b1a6/packages/static-module-record/src/hidden.js#L1-L2

**Comment by kriskowal** [comment, marked as answer]
The module to program translation works only if none of the free variables of the module collide with these names that the calling convention introduces into their scope. It is our hope that no real modules are incompatible with the ses shim because they were bound to these names in global scope.

JavaScript borrowed the idea of allowing generated code to include $ in compiler-generated names from Java. jQuery ensured that we would never be able to rely on the glyphs’s specialness for the intended purpose. Using an invisible code point at least minimizes the displacement of column numbers when debugging.

These choices may have been extreme.
